### ISSUE TITLE
state dict key error while loading pretrained rm

### ISSUE THREAD

**Issue by DaehanKim** [Dahoas/reward-modeling#3]
Thank you for sharing this work!

I found following README scripts to load pretrained rm weights leads to key errors.
I tried both `Dahoas/pythia-6b-rm-synthetic` and `Dahoas/gptj-rm-static`.
First one has both missing keys and unexpected keys; Second one has only unexpected keys.
So I guess I should be using `Dahoas/gptj-rm-static`.

This may be because `Dahoas/gptj-rm-static` weight has duplicate keys starting with "model" and without "model" flag.
<img width="823" alt="image" src="https://user-images.githubusercontent.com/20675681/233033557-f7a6ac30-9e22-4b4e-bbd5-61f11e935a19.png">

If this was intended, can you explain how I can load this weight properly? Thanks!

**Comment by Dahoas** [comment]
Sorry about this, indeed `Dahoas/pythia-6b-rm-synthetic` has duplicate keys. Can you try loading as the readme suggests:

```python
      import torch
      from utils import make_rm
      # save_model is used to determine whether a reference to the base model is saved in the RM wrapper (this is necessary to use HF's Activation Checkpointing code)
      save_model = False
      rm = make_rm("EleutherAI/gpt-j-6B", "causal", "EleutherAI/gpt-neox-20b", save_model)
      rm.load_state_dict(torch.load(PATH_TO_CKPT), strict=True)
```

to account for the duplicate keys set `save_model = True` when initializing the RM.
